### PR TITLE
[IMP] product: pricelist flow

### PR DESCRIPTION
**Pull request by Feyensv**
Pricelist flow isn't quite intuitive in its current state (and certainly not clean in a technical point of view).

# Improvements
1. Price Rules are opened in independent view from the product/product template.
2. Price rules of a product are still accessible in advanced pricelist mode.
  * From a product.template : You see the rules on this product and on its variants
  * From a product.product : You see the rules on the variant and on its template
3. UI refactorings

## Other changes
* Don't create a rule for all pricelist by default.  If no rule is defined on a pricelist for a given product, it takes the sales price by default.
* Technical : multiple overly complicated pricelist settings are removed.

[Task 34788](https://www.odoo.com/web?#id=34788&action=333&active_id=1725&model=project.task&view_type=form&menu_id=4720)

### CODE REVIEW

**Comment by Yenthe666** [comment]
Ooo that is an interesting concept. Would you like us to test it functionally @Feyensv?

**Comment by Feyensv** [comment]
@Yenthe666 I mainly wanted to know if you knew any flow/situation I removed through those changes (the objective is to simplify/clarify the flow, in a functional and technical point of view, without removing functionalities), but if you have any functional feedback to give, I'm also interested :).

File: addons/product/views/product_pricelist_views.xml [hunk 1]
```diff
@@ -23,61 +92,80 @@
...
+                              <!-- use date period picker? -->
+                          </group>
+                          <group name="pricelist_rule_related" groups="base.group_no_one">
+                              <!-- Infos from the pricelist for UI rendering (monetary fields, ...) -->
+                              <field name="pricelist_id" invisible="1"/>
+                              <field name="currency_id" groups="base.group_multi_currency"/>
+                              <field name="company_id" groups="base.group_multi_company"/>
+                          </group>
+                      </group>
+                      <group string="Price Computation" groups="product.group_sale_pricelist">
```
Comment (Yenthe666) [inline addons/product/views/product_pricelist_views.xml]: Might be nice to give this group a name too.

File: addons/product/views/product_pricelist_views.xml [hunk 2]
```diff
@@ -23,61 +92,80 @@
...
+                              <field name="base_pricelist_id" attrs="{
+                                'invisible': ['|', ('compute_price', '!=', 'formula'), ('base', '!=', 'pricelist')],
+                                'required': [('compute_price', '=', 'formula'), ('base', '=', 'pricelist')],
+                                'readonly': [('base', '!=', 'pricelist')]}"/>
+                          </group>
+                      </group>
+                      <div class="oe_grey" groups="uom.group_uom">
+                          <p>The computed price is expressed in the default Unit of Measure of the product.</p>
+                      </div>
+                      <group col="6" attrs="{'invisible':[('compute_price', '!=', 'formula')]}" groups="product.group_sale_pricelist">
```
Comment (Yenthe666) [inline addons/product/views/product_pricelist_views.xml]: group name please :)

File: addons/product/views/product_pricelist_views.xml [hunk 3]
```diff
@@ -142,28 +230,44 @@
                             <h1><field name="name" placeholder="e.g. USD Retailers"/></h1>
                         </div>
                         <group>
-                            <field name="discount_policy" groups="product.group_discount_per_so_line" widget="radio"/>
-                            <field name="currency_id" groups="base.group_multi_currency"/>
-                            <field name="active" invisible="1"/>
-                            <field name="company_id" groups="base.group_multi_company" options="{'no_create': True}"/>
-                            <field name="country_group_ids"/>
+                          <group>
```
Comment (Yenthe666) [inline addons/product/views/product_pricelist_views.xml]: also a group name please :)

**Comment by Yenthe666** [comment]
I did some tests and what I tested worked fine. I've added three suggestions for group names but besides of this it looks pretty nice! I could feel the simplification when using it :)

File: addons/product/models/product.py [hunk 4]
```diff
@@ -581,6 +580,25 @@ def view_header_get(self, view_id, view_type):
...
+            'name': _('Price Rules'),
+            'view_mode': 'tree,form',
+            'views': [(self.env.ref('product.product_pricelist_item_tree_view_from_product').id, 'tree'), (False, 'form')],
+            'res_model': 'product.pricelist.item',
+            'type': 'ir.actions.act_window',
+            'target': 'current',
+            'domain': domain,
+            'context': {
+                'default_product_id': self.id,
+                'default_applied_on': '0_product_variant',
```
[the remark this answers is not in the document]
Comment (Feyensv) [reply addons/product/models/product.py]: Nope, self is a product.product = product variant ;)

File: addons/product/models/product.py [hunk 5]
```diff
@@ -317,12 +316,12 @@ def _compute_product_template_attribute_value_ids(self):
                 else:
                     product.product_template_attribute_value_ids += values_per_template[product.product_tmpl_id.id][pav.id]
 
-    def _get_pricelist_items(self):
+    def _compute_variant_item_count(self):
```
[the remark this answers is not in the document]
Comment (Feyensv) [reply addons/product/models/product.py]: Well, it is possible, but we only see this info in the product form view.  Do we really need to optimize that far? :D

File: addons/product/views/product_pricelist_views.xml [hunk 6]
```diff
@@ -23,61 +92,80 @@
...
+                              <field name="product_id" attrs="{
+                                'invisible':[('applied_on', '!=', '0_product_variant')],
+                                'required':[('applied_on', '=', '0_product_variant')]}"
+                                options="{'no_create':1}"/>
+                          </group>
+                          <group name="pricelist_rule_limits">
+                              <field name="min_quantity"/>
+                              <field name="date_start"/>
+                              <field name="date_end"/>
+                              <!-- use date period picker? -->
```
[the remark this answers is not in the document]
Comment (Feyensv) [reply addons/product/views/product_pricelist_views.xml]: I thought I had removed this comment :D.

File: addons/product/models/product.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (dbo-odoo) [reply addons/product/models/product.py]: alright

**Comment by dbo-odoo** [comment]
@robodoo r+ rebase-merge

**Comment by dbo-odoo** [comment]
@robodoo r+ rebase-merge

**Comment by d-fence** [comment]
robodoo retry

**Pull request closed without merging**